xdsl/parser.py

```diff
@@ -1233,6 +1233,82 @@ def try_parse_builtin_attr(self) -> Attribute | None:
             if (val := attr_parser()) is not None:
                 return val
 
+        self._synchronize_lexer_and_tokenizer()
+        if self._current_token.text == "strided":
+            strided = self.parse_strided_layout_attr()
+            self._synchronize_lexer_and_tokenizer()
+            return strided
+
+        return None
+
+    def _parse_int_or_question(self,
```

Comment:
This feels like a rather particular function. Does it make sense to have this? Wouldn't a parametrized version make more sense?

Reply:
This is not really used anywhere else, which is why I made this private for now.
The other solution is to make this a nested function, but I don't really like it, what do you think?